homeassistant/components/zha/light.py

```diff
@@ -224,17 +228,51 @@ async def async_turn_on(self, **kwargs):
         effect = kwargs.get(light.ATTR_EFFECT)
         flash = kwargs.get(light.ATTR_FLASH)
 
-        if brightness is None and self._off_brightness is not None:
+        # If the light is currently off but a turn_on call with a color/temperature is sent,
+        # the light needs to be turned on first at a low brightness level where the light is immediately transitioned
+        # to the correct color. Afterwards, the transition is only from the low brightness to the new brightness.
+        # Otherwise, the transition is from the color the light had before being turned on to the new color.
+        # This can look especially bad with transitions longer than a second.
+        color_provided_from_off = (
+            not self._state
+            and brightness_supported(self._attr_supported_color_modes)
+            and (light.ATTR_COLOR_TEMP in kwargs or light.ATTR_HS_COLOR in kwargs)
+        )
+        final_duration = duration
+        if color_provided_from_off:
+            # Set the duration for the color changing commands to 0.
+            duration = 0
```

Comment:
Setting this to 0 will cause some lights to ignore color commands. IIRC some sengled pro models refuse to change temp without a transition. That’s why the sentinel default is 1.

Reply:
Setting it to ``1`` (0.1 seconds) is still better than the current behavior -- it's not as nice as ``0`` for lights that support it though. I find it tolerable though.

But I do recall that there's something with IKEA lights not accepting further commands during transitions (?) and I wonder if that's going to be an issue. I'll definitely make sure to test those lights.

(Also, maybe there should be a separate config option for fully enabling/disabling this color transition behavior and/or setting that ``0`` or ``1`` as a transition time for the color changes? But that also seems "technical", so I wouldn't really want to expose that transition setting (which is "rarely" used) to the user)

(I'll also search for the comments on the Sengled Pro lights. Maybe we're lucky and a transition > 0 is only required for light levels and not color transitions haha)

Reply:
Otherwise, if just Sengled lights are affected, we could make another constant like ``DEFAULT_COLOR_TRANSITION = 0`` which gets overridden to ``1`` for all Sengled lights (similar to how the ``force_on`` is done for ``Jasco``).

Then, there would be "perfect behavior" for every bulb which follows specs, the IKEA transition stuff wouldn't be an issue (but should still be tested), and a config option for enabling/disabling "this feature" and/or setting that specific transition could be avoided.

Reply:
It was color temperature that was affected for the sengled lights. The default is already configurable in the ZHA config panel.

Reply:
https://github.com/home-assistant/core/blob/4f7c365a0b8d10bccfda2b9856250748864ff447/homeassistant/components/zha/light.py#L221

Reply:
That's a transition a user might want to see though. We want no "transition" here if possible (for the color changing part). The fading up with the light level should occur at the normal level. (In the linked video, there's only one frame of the light showing the old red color with 0 transition for the color and that's good. The level fade up when turning on has the expected transition.)

The user-configured value could be set to 1 second (or longer) for example and it would break all of this.

(It also can't be set to 0, as that's going to use the default 0.1 second default and other values like 0.5 aren't possible with it at the moment.)